util/session/sessionmanager.go

```diff
@@ -268,19 +268,26 @@ func (mgr *SessionManager) Parse(tokenString string) (jwt.Claims, string, error)
 		return nil, "", fmt.Errorf("account password has changed since token issued")
 	}
 
+	exp, err := jwtutil.ExpirationTime(claims)
+	if err != nil {
+		return nil, "", fmt.Errorf("token has no expiration time: %s", err)
+	}
```

Comment:
Throwing error if token does not have expiration time set.

Reply:
Is this an absolutely necessary change? If the user is directly using an OIDC provider instead of going through Dex, I could imagine the provider sending non-expiring tokens, and the user wanting to retain that functionality.

Reply:
@crenshaw-dev The expiration time check is strictly related to Admin(Basic authentication) tokens created by ArgoCD itself. With the current check and https://github.com/argoproj/argo-cd/pull/15004#discussion_r1349437286 I am preventing ArgoCD to indefinitely re-up a login session.

Reply:
@crenshaw-dev To the your concern about Dex or OIDC tokens, they are checked in different section: https://github.com/argoproj/argo-cd/blob/1600c03abafc4a0526d6909b3eddab5071d59508/util/session/sessionmanager.go#L520-L557
and more specific: https://github.com/argoproj/argo-cd/blob/1600c03abafc4a0526d6909b3eddab5071d59508/util/session/sessionmanager.go#L535
if you follow it to the bottom you will see that ArgoCD does not tolerate tokens without expiration time because this flag: https://github.com/coreos/go-oidc/blob/3ebd0c9e4dcfceac8e1b23b1fb5ff8e7b8b61c2a/oidc/verify.go#L285 is not exposed to a user and default value is "false".